src/compiler/transformers/declarations.ts

```diff
@@ -973,10 +973,124 @@ namespace ts {
                             input.isTypeOf
                         ));
                     }
+                    case SyntaxKind.UnionType: {
+                        if(input.types.length <=1) return cleanup(input);
+                        const clone = getMutableClone(input);
+                        const types = clone.types.map(n=>n as TypeReferenceNode);

```

Comment:
I think `.slice()` is the usual way to copy an array. Is copying necessary here, since this is a mutable clone?